tests/prototypes/prototype_kafka.py

```diff
@@ -0,0 +1,395 @@
+
+    def __del__(self) -> None:
+        if kafka_topic_exists(self.kafka_topic, self.admin):
+            delete_topic(self.kafka_topic, self.admin)
+
+
+class KafkaStreamSimulatorScippOnly(KafkaStreamSimulatorBase):
+    random_events: RandomEvents
+
+    def stream(self) -> None:
+        import json
+
+        from scipp.serialization import serialize
+
+        for random_event in self.random_events:
+            header, serialized_list = serialize(random_event)
+            self.producer.produce(self.kafka_topic, f'header:{json.dumps(header)}')
+
+            for data_buffer in serialized_list:
+                self.produce_data(data_buffer)
+                self.producer.produce(self.kafka_topic, 'finished')
+                self.debug("Result: %s", self.producer.poll(1))
+
+
+class KafkaStreamSimulator(KafkaStreamSimulatorBase):
+    random_events: RandomEventBuffers
+
+    def stream(self) -> None:
+        for i_frame, random_event in enumerate(self.random_events):
+            self.producer.produce(self.kafka_topic, "starts")
+            self.produce_data(random_event)
+            self.producer.produce(self.kafka_topic, 'finished')
+            self.debug(
+                "Produced %sth message divided into %s chunks.",
+                i_frame + 1,
+                self.producer.poll(1),
+            )
+
+
+class KafkaListenerBase(BaseApp, ABC):
+    raw_data_pipe: List[Events]
+    chunk_size: ChunkSize
+    kafka_topic: KafkaTopic
+    consumer_cxt: ConsumerContextManager
+    num_frames: NumFrames
+    frame_rate: FrameRate
+
+    def merge_bytes(self, datalist: list[bytes]) -> bytes:
+        from functools import reduce
+
+        return reduce(lambda x, y: x + y, datalist)
+
+    @retry(Empty, max_trials=10, interval=0.1)
+    def _poll(self, consumer: Consumer) -> Message:
+        if (msg := consumer.poll(timeout=0)) is None:
+            raise Empty
+        else:
+            return msg
+
+    def poll(self, consumer: Consumer) -> Message | None:
+        try:
+            return self._poll(consumer)
+        except Empty:
+            return None
+
+    @abstractmethod
+    def poll_one_data(self, consumer: Consumer) -> sc.DataArray | None:
+        ...
+
+    async def send_data_chunk(self, data_chunk: Events, i_frame: int) -> None:
+        import asyncio
+
+        self.debug("Sending %s th, %s pieces of data.", i_frame + 1, len(data_chunk))
+        self.raw_data_pipe.append(Events(data_chunk))
+        await asyncio.sleep(0)
+
+    def start_stop_watch(self) -> None:
+        self.stop_watch.start()
+
+    async def run(self) -> None:
+        with self.consumer_cxt() as consumer:
+            self.start_stop_watch()
+            data_chunk: Events = Events([])
+            i_frame = 0
+            while (event := self.poll_one_data(consumer)) is not None:
+                data_chunk.append(event)
+                if len(data_chunk) >= self.chunk_size and (i_frame := i_frame + 1):
```

Comment:
No, it was just to bind it with the condition `len(data_chunk) >= self.chunk_size` .... so that I don't forget to change `i_frame` as well if the condition changes.
Is it more confusing than convenient....?

Reply:
> Is it more confusing than convenient....?

Maybe?